library/src/main/java/com/google/android/exoplayer/text/subrip/SubripParser.java

```diff
@@ -0,0 +1,149 @@
+
+  private static final String SUBRIP_CUE_IDENTIFIER_STRING = "^(.*)\\s-->\\s(.*)$";
+  private static final Pattern SUBRIP_CUE_IDENTIFIER =
+          Pattern.compile(SUBRIP_CUE_IDENTIFIER_STRING);
+
+  private static final String SUBRIP_TIMESTAMP_STRING = "(\\d+:)?[0-5]\\d:[0-5]\\d:[0-5]\\d,\\d{3}";
+  // private static final Pattern SUBRIP_TIMESTAMP = Pattern.compile(SUBRIP_TIMESTAMP_STRING);
+
+  private final StringBuilder textBuilder;
+
+  private final boolean strictParsing;
+
+  public SubripParser() {
+    this(true);
+  }
+
+  public SubripParser(boolean strictParsing) {
+    this.strictParsing = strictParsing;
+
+    textBuilder = new StringBuilder();
+  }
+
+  @Override
+  public SubripSubtitle parse(InputStream inputStream, String inputEncoding, long startTimeUs)
+          throws IOException {
+    ArrayList<SubripCue> subtitles = new ArrayList<>();
+
+    // file should not be empty
+    if (inputStream.available() == 0) {
+      throw new ParserException("File is empty?");
+    }
+
+    BufferedReader subripData = new BufferedReader(new InputStreamReader(inputStream, C.UTF8_NAME));
+    String line;
+
+
+    // process the cues and text
+    while ((line = subripData.readLine()) != null) {
+      long startTime = Cue.UNSET_VALUE;
+      long endTime = Cue.UNSET_VALUE;
+      CharSequence text = null;
+      int position = Cue.UNSET_VALUE;
+
+      Matcher matcher = SUBRIP_POSITION.matcher(line);
+      if (matcher.matches()) {
+        position = Integer.parseInt(matcher.group());
+      }
+
+      line = subripData.readLine();
+
+      // parse cue time
+      matcher = SUBRIP_CUE_IDENTIFIER.matcher(line);
+      if (!matcher.find()) {
+        throw new ParserException("Expected cue start time: " + line);
+      } else {
+        startTime = parseTimestampUs(matcher.group(1)) + startTimeUs;
+        endTime = parseTimestampUs(matcher.group(2)) + startTimeUs;
+      }
+
+      // parse text
+      textBuilder.setLength(0);
+      while (((line = subripData.readLine()) != null) && (!line.isEmpty())) {
+        if (textBuilder.length() > 0) {
+          textBuilder.append("<br>");
+        }
+        textBuilder.append(line.trim());
+      }
+      text = Html.fromHtml(textBuilder.toString());
+
+      SubripCue cue = new SubripCue(startTime, endTime, position, text);
+      subtitles.add(cue);
+    }
+
+    subripData.close();
+    inputStream.close();
+    SubripSubtitle subtitle = new SubripSubtitle(subtitles, startTimeUs);
+    return subtitle;
+  }
+
+  @Override
+  public boolean canParse(String mimeType) {
+    return MimeTypes.APPLICATION_SUBRIP.equals(mimeType);
+  }
+
+  protected void handleNoncompliantLine(String line) throws ParserException {
+    if (strictParsing) {
+      throw new ParserException("Unexpected line: " + line);
+    }
+  }
+
+  private static long parseTimestampUs(String s) throws NumberFormatException {
+    if (!s.matches(SUBRIP_TIMESTAMP_STRING)) {
```

Comment:
Normally we'd use a Matcher for this, I think? Like you do for SUBRIP_POSITION. Any reason why not to?